### DISCUSSION TITLE
Chaining without burn?  Simple question - Complex answer I'm sure

### DISCUSSION THREAD

**Discussion by mclark114** [wixtoolset/issues#6483] [Questions]
I have a customer that  wants to install vs_redistx64.exe without creating and executables. I have seen in commercial Installers they provide this ability. 
Can this be done in Wix ?

Thanks
-Michael

**Comment by BMurri** [comment]
In theory, yes, the same way its done in those commercial installers.

Here are the issues:
- Windows Installer does not support serial installations (with the exception of chained MSIs most easily accessed via Embedded UI, neither of which are easy to use, and both of which have severe functional limitations, critical ones of which are not publicly disclosed by MSFT).
- Nested installations, which have been officially deprecated for decades now, would produce severe usability problems (but that doesn't work for exe installers anyway).
- The way that the commercial installers provide this functionality is by calling the prerequisite installer(s) via a custom action from the UI sequence. If the user launches the MSI with the UI suppressed, the prereqs will NOT be installed. If the MSI itself requires elevation, the user will be prompted twice for elevation. The workaround for both of those issues is to wrap the MSI with an EXE that requires elevation.

Given both all the problems of MSI UI and the above issues, the modern best practice is to provide your installation UI in an executable that manages one or more MSI packages. Prerequisites that ship via their own executables is supported (by necessity) by all the viable bootstrappers/chainers on the market, including WiX.
